sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-file.service.ts

```diff
@@ -0,0 +1,161 @@
+import { Injectable } from "@nestjs/common";
+import { DisbursementSchedule } from "@sims/sims-db";
+import { LoggerService, InjectLogger } from "@sims/utilities/logger";
+import {
+  ConfigService,
+  InstitutionIntegrationConfig,
+} from "@sims/utilities/config";
+import { getFileNameAsCurrentTimestamp } from "@sims/utilities";
+import { DisbursementScheduleService } from "@sims/integrations/services";
+import { ECERecord, ECEUploadResult } from "./models/ece-integration.model";
+import { ECEIntegrationService } from "./ece-integration.service";
+
+@Injectable()
+export class ECEFileService {
+  institutionIntegrationConfig: InstitutionIntegrationConfig;
+  constructor(
+    config: ConfigService,
+    private readonly eceIntegrationService: ECEIntegrationService,
+    private readonly disbursementScheduleService: DisbursementScheduleService,
+  ) {
+    this.institutionIntegrationConfig = config.institutionIntegration;
+  }
+
+  /**
+   * 1. Fetches the applications that have an eligible COE request waiting.
+   * 2. Create the Request content for the ECE request file by populating the
+   * header, footer and trailer content.
+   * 3. Create the request filename with the file path with respect to the institution code
+   * for the ECE request sent File.
+   * 4. Upload the content to the zoneB SFTP server.
+   * @param generatedDate date in which the eligible coe for
+   * particular institution is required.
+   * @returns Processing ECE request result.
+   */
+  async processECEFile(generatedDate?: string): Promise<ECEUploadResult[]> {
+    this.logger.log(`Retrieving eligible COEs for ECE request...`);
+    const eligibleCOEs = await this.disbursementScheduleService.getEligibleCOE(
+      generatedDate,
+    );
+    if (!eligibleCOEs.length) {
```

Comment:
This one used the Typerm `find` that should be the same from `getMany` that should return an empty array (please double check @guru-aot).
I guess that our method `getRawAndEntities` is not following it @ann-aot, but it should. Instead of returning `null` it should return an empty array also.

Reply:
Yes an empty array only will be returned, if no rows are returned. typeorm find(options?: FindManyOptions<Entity>): Promise<Entity[]>; will return an Entity array as a promise, so it will never have null or undefined.